src/wallet/spend.cpp

```diff
@@ -46,17 +90,65 @@ int CalculateMaximumSignedInputSize(const CTxOut& txout, const CWallet* wallet,
     return CalculateMaximumSignedInputSize(txout, COutPoint(), provider.get(), coin_control);
 }
 
+/** Infer a descriptor for the given output script. */
+static std::unique_ptr<Descriptor> GetDescriptor(const CWallet* wallet, const CCoinControl* coin_control,
+                                                 const CScript script_pubkey)
+{
+    if (const auto provider = wallet->GetSolvingProvider(script_pubkey)) {
+        return InferDescriptor(script_pubkey, *provider);
+    }
+    if (coin_control) {
+        return InferDescriptor(script_pubkey, coin_control->m_external_provider);
+    }
```

Comment:
Hmm good point. Unfortunately this is a bit clumsy for the `LegacyScriptPubKeyMan`: since it's not a `FlatSigningProvider` it involves copying all its data in order to merge both providers. I've pushed the change in a separate commit to be squashed if it's not too unreasonable. What do you think?

Reply:
> What do you think?

@achow101 followup thoughts here?

Reply:
Pushed a much cleaner version of this: introducing a `MultiSigningProvider` to wrap around both the wallet and external providers. Still as a commit on top, let me know what you think @achow101.